src/mercury_engine_data_structures/formats/property_enum.py

```diff
@@ -6,15 +7,32 @@
 from mercury_engine_data_structures import crc, dread_data
 
 
+class HashSet(enum.Enum):
+    DREAD_PROPERTY = enum.auto()
+    DREAD_FILE_NAME = enum.auto()
+
+    def get_hashes(self) -> tuple[Dict[str, int], Dict[int, str]]:
+        if self == HashSet.DREAD_PROPERTY:
+            return dread_data.all_name_to_property_id(), dread_data.all_property_id_to_name()
+        elif self == HashSet.DREAD_FILE_NAME:
+            return dread_data.all_name_to_asset_id(), dread_data.all_asset_id_to_name()
+        else:
+            raise ValueError("Unknown")
+
+
 class CRCAdapter(construct.Adapter):
-    def __init__(self, subcon, known_hashes: Dict[str, int], allow_unknowns=False, display_warnings=True):
-        super().__init__(subcon)
-        self.known_hashes = known_hashes
-        self.inverted_hashes = {value: name for name, value in known_hashes.items()}
+    known_hashes: Dict[str, int]
+    inverted_hashes: Dict[int, str]
+
+    def __init__(self, hash_set: HashSet, allow_unknowns=False, display_warnings=True):
+        super().__init__(construct.Hex(construct.Int64ul))
+        self._raw_subcon = construct.Int64ul
```

Comment:
should these be hard-coded? I thought MSR used crc32 but I may be mistaken

Reply:
I'll fix that when we this gets used. There's also a `crc64` call somewhere

Reply:
dcr *is* actively working on stuff, it'd probably be nice to do it in advance for him

Reply:
This is going to be significantly harder as it'd be nice to make `PropertyEnum` (and others) detect which game it is, instead of having to duplicate into `DreadPropertyEnum` and `SRPropertyEnum`, which is how it's designed right now.

Gonna leave that for another PR.